File: Code/Server/Revenj.Processing/CommandResultDescription.cs
using System;
using System.Diagnostics;

namespace Revenj.Processing
{
	public class CommandResultDescription<TFormat> : ICommandResultDescription<TFormat>
	{
		public string RequestID { get; private set; }
		public ICommandResult<TFormat> Result { get; private set; }
		public long Duration { get; private set; }

		public static CommandResultDescription<TFormat> Create(string id, ICommandResult<TFormat> result, long start)
		{
			return new CommandResultDescription<TFormat>
			{
				RequestID = id,
				Result = result,
				Duration = (Stopwatch.GetTimestamp() - start) / TimeSpan.TicksPerMillisecond
			};
		}
	}
}

File: java/revenj-servlet/src/main/java/org/revenj/server/CommandResultDescription.java
package org.revenj.server;

public final class CommandResultDescription<TFormat> {
	public final String requestID;
	public final CommandResult<TFormat> result;
	public final long duration;

	CommandResultDescription(String requestID, CommandResult<TFormat> result, long start) {
		this.requestID = requestID;
		this.result = result;
		duration = (System.nanoTime() - start) / 1_000;
	}

	public static <TFormat> CommandResultDescription<TFormat> create(String requestID, CommandResult<TFormat> result, long start) {
		return new CommandResultDescription<>(requestID, result, start);
	}
}
